### PR TITLE
Feature: setup step for zkEngine that produces PP

### PR DESCRIPTION
**Pull request by Forpee**
I added a `setup()` step to the main structures for proving (`ZKEProof` & `Batched ZKEProof`). Previously, this setup was done inside the proving structures’ `prove_wasm` function.

This `setup()` function must now be run by both the prover and the verifier before they proceed with proving or verifying.

For more details, you can refer to the updated examples and README, which illustrate how this step is integrated into the proving/verifying flow.

**Additional changes:**

- Removed K3 tests (they are now maintained in the K3 repository).
- Cleaned up the default MCC implementation, which was previously performing unnecessary commitments.
- Removed the ZKML example, as it relied on the WASI random function (`(import "wasi_snapshot_preview1" "random_get" (func (;2;) (type 0)))`). This caused issues where the verifier and prover would receive different random values during the setup step, making it impossible to prove WASM when `PP` is produced at different times.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -33,60 +33,38 @@ RUST_LOG=debug cargo +nightly run --release --example default
...
+  let pp = ZKEngine::setup(&mut WASMCtx::new_from_file(&args)?)?;
+
+  // Prove execution and run memory consistency checks
+  //
+  // Get proof for verification and corresponding public values
+  //
+  // Above type alias's (for the backend config) get used here
+  let (proof, public_values, _) = ZKEngine::prove_wasm(&mut WASMCtx::new_from_file(&args)?, &pp)?;
+
+  // // Verify proof
```
Comment (altonen) [inline README.md:64]:
```suggestion
  // Verify proof
```

File: examples/default.rs [hunk 2]
```diff
@@ -34,18 +16,17 @@ fn main() -> anyhow::Result<()> {
...
   // Get proof for verification and corresponding public values
   //
   // Above type alias's (for the backend config) get used here
-  let (proof, public_values, _) =
-    ZKEProof::<E1, BS1<E1>, S1<E1>, S2<E1>>::prove_wasm(&mut wasm_ctx)?;
+  let (proof, public_values, _) = ZKEngine::prove_wasm(&mut WASMCtx::new_from_file(&args)?, &pp)?;
 
-  // Verify proof
-  let result = proof.verify(public_values)?;
+  // // Verify proof
```
Comment (altonen) [inline examples/default.rs:29]:
```suggestion
  // Verify proof
```

File: examples/serialized.rs [hunk 3]
```diff
@@ -39,19 +39,22 @@ fn main() -> anyhow::Result<()> {
...
   // Deserialize the proof and public values
   let proof: ZKEProof<E1, BS1, S1, S2> = serde_json::from_str(&proof_str)?;
-  let public_values: PublicValues<E1, BS1, S1, S2> = serde_json::from_str(&public_values_str)?;
+  let public_values: PublicValues<E1> = serde_json::from_str(&public_values_str)?;
 
-  // Verify the proof
-  let result = proof.verify(public_values)?;
+  // Verify proof
+  let pp = ZKEngine::setup(&mut WASMCtx::new_from_file(&args)?)?;
+  let result = proof.verify(public_values, &pp)?;
```
Comment (altonen) [inline examples/serialized.rs:58]: dq: how is `pp` different from `public_values`? I checked locally and bulk of the data is now in `pp` which is preferable but what does `public_values` contain at this point?

**Review by altonen — APPROVED** [review]

File: examples/serialized.rs (hunk 3 above)
Comment (Forpee) [reply examples/serialized.rs:58]: Public values just contains the i/o of the proving system which is just 2 vecs of scalars values (one each for i/o )which are not very long/big.

File: README.md (hunk 1 above)
Comment (Forpee) [reply README.md]: thanks

**Comment by altonen** [comment]
wen merge & companion?

**Comment by Forpee** [comment]
@altonen doing companion PR now

**Pull request merged**